assets/chat/js/chat.js

```diff
@@ -1319,29 +1329,78 @@ class Chat {
     }
 
     cmdIGNORE(parts){
-        const username = parts[0] || null;
-        if (!username) {
+        if (!parts[0]) {
             if (this.ignoring.size <= 0) {
                 MessageBuilder.info('Your ignore list is empty').into(this);
             } else {
                 MessageBuilder.info(`Ignoring the following people: ${Array.from(this.ignoring.values()).join(', ')}`).into(this);
             }
-        } else if (!nickregex.test(username)) {
-            MessageBuilder.info('Invalid nick - /ignore <nick>').into(this);
+        } else if (parts.length > 1) {
+            // this is a little ugly, but it allows us to not ignore anything if there's an invalid nick in there
+            // think that's less confusing/nicer compared to partially ignoring 
+            let validUsernames = new Set();
+            // .some() stops iterating over the array if the inner function returns true
+            // which is perfect for our use case
+            const failure = parts.some(username => {
+                if (!nickregex.test(username)) {
+                    MessageBuilder.info(`${username} is not a valid nick - /ignore <nick> OR /ignore <nick_1> <nick_2> ... <nick_n>`).into(this);
+                    return true;
+                } else {
+                    validUsernames.add(username);
+                    return false;
+                }
+            });
+            if (!failure) {
+                validUsernames.forEach(username => {
+                    this.ignore(username, true);
+                    this.removeMessageByNick(username);
+                });
+                MessageBuilder.status(`Added the following people to your ignore list: ${Array.from(validUsernames.values()).join(', ')}`).into(this);
+            }
         } else {
```

Comment:
Rather than processing the command differently if there's only one username, can we simplify things here and process it the same way whether it's one or many?

Reply:
Oh, true, fixed it up